services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/overviewSection.tsx

```diff
@@ -1,96 +1,256 @@
+    }
+  }, [preDiagnosticAssignedData, postDiagnosticAssignedData, preDiagnosticCompletedData, postDiagnosticCompletedData, recommendationsData])
+
+  function initializePusher() {
+    pusherChannel?.bind(PUSHER_EVENT_KEY, (body) => {
+      const { message, } = body
+
+      setPusherMessage(message)
+    });
+  };
+
+  function getSearchParams(queryString) {
+    return {
+      query: queryString,
+      timeframe: selectedTimeframe,
+      school_ids: selectedSchoolIds,
+      teacher_ids: selectedTeacherIds,
+      classroom_ids: selectedClassroomIds,
+      grades: selectedGrades,
+      group_by: groupByValue.value
+    }
+  }
+
+  function getData() {
+    setLoading(true)
+    getPreDiagnosticAssignedData()
+    getPreDiagnosticCompletedData()
+    getPostDiagnosticAssignedData()
+    getPostDiagnosticCompletedData()
+    getRecommendationsData()
+  }
+
+  function getPreDiagnosticAssignedData () {
+    const searchParams = getSearchParams("pre-diagnostic-assigned")
+
+    requestPost('/admin_diagnostic_reports/report', searchParams, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        return
+      } else {
+        const { results, } = body
+        setPreDiagnosticAssignedData(results)
+      }
+    })
+  }
+
+  function getPostDiagnosticAssignedData() {
+    const searchParams = getSearchParams("post-diagnostic-assigned")
+
+    requestPost('/admin_diagnostic_reports/report', searchParams, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        return
+      } else {
+        const { results, } = body
+        setPostDiagnosticAssignedData(results)
+      }
+    })
+  }
+
+  function getPreDiagnosticCompletedData() {
+    const searchParams = getSearchParams("pre-diagnostic-completed")
+
+    requestPost('/admin_diagnostic_reports/report', searchParams, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        return
+      } else {
+        const { results, } = body
+        setPreDiagnosticCompletedData(results)
+      }
+    })
+  }
+
+  function getPostDiagnosticCompletedData() {
+    const searchParams = getSearchParams("post-diagnostic-completed")
+
+    requestPost('/admin_diagnostic_reports/report', searchParams, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        return
+      } else {
+        const { results, } = body
+        setPostDiagnosticCompletedData(results)
+      }
+    })
+  }
+
+  function getRecommendationsData() {
+    const searchParams = getSearchParams("recommendations")
+
+    requestPost('/admin_diagnostic_reports/report', searchParams, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        return
+      } else {
+        const { results, } = body
+        setRecommendationsData(results)
+      }
+    })
+  }
+
+  function filtersMatchHash(hashMessage) {
+    const filterTarget = [].concat(
+      QUERY_KEY,
+      selectedTimeframe,
+      selectedSchoolIds,
+      selectedGrades,
+      selectedTeacherIds,
+      selectedClassroomIds
```

Comment:
So this isn't currently in the back-end, but reviewing this suggests to me that we should also have `groupByValue` in here.  Let's just do that in a separate PR, though, to avoid adding complexity to this one.